### PR TITLE
H-1028: Disable realtime service by default

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

The `realtime` service reads database changes from a replica slot and adds them to a Redis queue, which can be subscribed to by other services which need to react to changes in the database. Previously, this included the collaborative editing ("collab") service, but currently it is only the service which syncs data changes in HASH back to external integrations.

With the introduction of authorization we need to figure out the permissions for this service. Until we do so, and until we return to developing external integrations, this PR disables `realtime` and the integration syncing function by default (it can be enabled by environment variable).

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: apps/hash-api/src/integrations/sync-back-watcher.ts [hunk 1]
```diff
@@ -40,38 +40,43 @@ export const createIntegrationSyncBackWatcher = async (
...
-      sendEntityToRelevantProcessor(entity, graphApiClient);
+        sendEntityToRelevantProcessor(entity, graphApiClient);
 
-      return true;
-    });
+        return true;
+      })
+      .catch((err) => {
+        // eslint-disable-next-line no-console -- caught because this function loses ownership of the queue occasionally in dev
+        console.error(`Could not take message from queue: ${err.message}`);
```
Comment (CiaranMn) [inline apps/hash-api/src/integrations/sync-back-watcher.ts:78]: Drive-by: the API would occasionally crash in dev if left for a while because it temporarily lost access to the Redis queue for some reason. I'm not sure why this is but am adding this here so that it doesn't tank the API, when realtime sync is enabled.

File: libs/@local/hash-backend-utils/src/environment.ts [hunk 2]
```diff
@@ -33,3 +33,5 @@ export const waitOnResource = async (
 
   logger?.debug(`${resource} is ready`);
 };
+
+export const realtimeSyncEnabled = !!process.env.ENABLE_REALTIME_SYNC;
```
Comment (benwerner01) [inline libs/@local/hash-backend-utils/src/environment.ts:37]: Maybe let's add `ENABLE_REALTIME_SYNC` to the committed `.env` and set it to `false` so it's easier to re-enable this in development down the line. Then we can modify this check to:

```suggestion
export const realtimeSyncEnabled = process.env.ENABLE_REALTIME_SYNC === "true";
```

File: apps/hash-api/src/integrations/sync-back-watcher.ts [hunk 3]
```diff
@@ -40,38 +40,43 @@ export const createIntegrationSyncBackWatcher = async (
   const queue = new RedisQueueExclusiveConsumer(redisClient);
 
   const processQueueMessage = () => {
-    void queue.pop(queueName, null, async (item: string) => {
-      const message = JSON.parse(item) as Wal2JsonMsg;
+    queue
```
Comment (benwerner01) [inline apps/hash-api/src/integrations/sync-back-watcher.ts:43]: Strange, I would've expected eslint to complain about `void` missing here. Is it not required when using `.catch`?


```suggestion
    void queue
```

**Review by benwerner01 — APPROVED** [review]
Thanks Ciaran, this is working for me locally. Left one small suggestion, but doesn't block merging.

File: apps/hash-api/src/integrations/sync-back-watcher.ts (hunk 3 above)
Comment (CiaranMn) [reply apps/hash-api/src/integrations/sync-back-watcher.ts:43]: Correct - the requirements are that one of the following list applies:
> Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator

File: libs/@local/hash-backend-utils/src/environment.ts (hunk 2 above)
Comment (CiaranMn) [reply libs/@local/hash-backend-utils/src/environment.ts]: Done in https://github.com/hashintel/hash/pull/3373/commits/8c4346cbc92e67f58ec0fb66fe62179a27327be6

**Review by benwerner01 — APPROVED** [review]
